applications/crossbar/src/modules/cb_port_requests.erl

```diff
@@ -636,6 +636,44 @@ read_descendants_fold(Account, Acc, Context) ->
             ]
     end.
 
+-spec maybe_read_descendant(cb_context:context(), ne_binary()) -> api_object().
+-spec maybe_read_descendant(cb_context:context(), ne_binary(), boolean()) -> api_object().
+maybe_read_descendant(Context, AccountId) ->
+    maybe_read_descendant(Context, AccountId, wh_services:is_reseller(AccountId)).
+
+maybe_read_descendant(Context, AccountId, 'true') ->
+    Authority =
+        case kz_whitelabel:fetch(AccountId) of
+            {'error', _R} -> 'undefined';
+            {'ok', JObj} ->
+                kz_whitelabel:port_authority(JObj)
+        end,
+    case Authority of
```

Comment:
This can probably be a function, used by the other function clause's `case Authority of` too.